addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2915,7 +2915,7 @@ const ImageHandlerOption = SnippetOptionWidget.extend({
             case 'selectWidth':
                 return img.naturalWidth;
             case 'setFilter':
-                return img.dataset.filter;
```

Comment:
`_computeWidgetState` returns the value which is currently used. By doing this, you say that all images have a filter by default, and that filter is previewed by the colorpicker widget. What you need to do is add a config option on the colorpicker widget to preselect a specific color if the current color is transparent when the colorpicker is being opened.